All: Updates for jQuery 1.12/2.2-3.5

mgol opened the pull request:
This has many changes but the ones on which I'd focus the most during reviews are:
* New categories for jQuery 3.5 but also older versions
* `even`/`odd` docs, removal of `:even`/`:odd` usage on API pages
* `addClass`/`removeClass`/`toggleClass`: array input signature + examples
* `htmlPrefilter`: rewritten, examples removed. The previous example of a more thorough check still wasn't covering all the cases so I didn't want people to have an impression this can be done securely like that.

Fixes #947
Fixes #949
Fixes #950
Fixes #1142
Ref #970
Ref #972

dmethvin commented on entries/even.xml:
```diff
@@ -0,0 +1,47 @@
+  <signature>
+    <added>3.5</added>
+  </signature>
+  <desc>Reduce the set of matched elements to the even ones in the set.</desc>
```
I know you mentioned it below but for people who don't read maybe `...in the set, numbered from zero.`

mgol replied:
Good point!

dmethvin commented on entries/jQuery.ajax.xml:
```diff
@@ -97,7 +97,7 @@ $.ajax({
             <li><code>"xml"</code>: Returns a XML document that can be processed via jQuery.</li>
             <li><code>"html"</code>: Returns HTML as plain text; included script tags are evaluated when inserted in the DOM.</li>
-            <li><code>"script"</code>: Evaluates the response as JavaScript and returns it as plain text. Disables caching by appending a query string parameter, <code>_=[TIMESTAMP]</code>, to the URL unless the <code>cache</code> option is set to <code>true</code>. <strong>Note:</strong> This will turn POSTs into GETs for remote-domain requests.</li>
+            <li><code>"script"</code>: Evaluates the response as JavaScript and returns it as plain text. Disables caching by appending a query string parameter, <code>_=[TIMESTAMP]</code>, to the URL unless the <code>cache</code> option is set to <code>true</code>. <strong>Note:</strong> This will turn POSTs into GETs for remote-domain requests. Prior to jQuery 3.5.0, unsuccessfull HTTP response ...
```
Just one `l` in "unsuccessful"

dmethvin commented on entries/jQuery.getScript.xml:
```diff
@@ -56,6 +56,7 @@ $( "div.log" ).ajaxError(function( e, jqxhr, settings, exception ) {
   }
 });
     </code></pre>
+    <p>Prior to jQuery 3.5.0, unsuccessfull HTTP responses with a script <code>Content-Type</code> were still executed.</p>
```
Just one `l` in "unsuccessful"

dmethvin commented on entries/odd.xml:
```diff
@@ -0,0 +1,47 @@
+  <signature>
+    <added>3.5</added>
+  </signature>
+  <desc>Reduce the set of matched elements to the odd ones in the set.</desc>
```
Same comment as with `even`,  maybe `...in the set, numbered from zero.`

mgol commented:
@dmethvin Thanks for feedback! [I addressed all your comments.](https://github.com/jquery/api.jquery.com/compare/1959289fd755799c0c7e4421efaa90e4b77876d5..03ea98efd98179a44c41126d6abc4a034bcb10ad)

timmywil commented on entries/jQuery.get.xml:
```diff
@@ -22,7 +22,7 @@
   </signature>
   <signature>
-    <added>1.12/2.2</added>
+    <added>1.12-2.2</added>
```
Is the dash necessary? This implies that all the versions between 1.12 and 2.2 are involved.

timmywil replied:
I understand the `/` doesn't work, but what about a `+` or space?

mgol replied:
Neither of these work, unfortunately. :( Neither does `_` or `,`. The checks there are pretty restrictive...

I went for `1.12-and-2.2`. Not pretty but perhaps clearer that what we have now.
